hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/util/TestRunJar.java

```diff
@@ -297,4 +297,35 @@ public void testUnJar2() throws IOException {
           "would create file outside of", e);
     }
   }
-}
\ No newline at end of file
+
+  /**
+   * Tests that RunJar errors appropriately for classes with non-static main
+   * methods.
+   */
+  public void testRunNonStaticMain() throws Throwable {
+    RunJar runJar = spy(new RunJar());
+    // enable the client classloader
+    when(runJar.useClientClassLoader()).thenReturn(true);
+    // set the system classes and blacklist the test main class and the test
+    // third class so they can be loaded by the application classloader
+    String mainCls = NonStaticMain.class.getName();
+    String systemClasses = "-" + mainCls + "," + ApplicationClassLoader.SYSTEM_CLASSES_DEFAULT;
+    when(runJar.getSystemClasses()).thenReturn(systemClasses);
+
+    // create the test jar
+    File testJar = JarFinder.makeClassLoaderTestJar(this.getClass(), TEST_ROOT_DIR, TEST_JAR_2_NAME, BUFF_SIZE,
+        mainCls);
+    // form the args
+    String[] args = new String[3];
```

Comment:
What is with [2]?

Reply:
I took this from `testClientClassLoader`.